tests/parsing/parchmint-parser.test.js

```diff
@@ -19,6 +19,127 @@ function parseJSONObj(str) {
     return JSON.parse('{' + str + '}');
 }
 
+describe('initialization', () => {
+    describe('set Parchmint text', () => {
+        test('constructor', () => {
+            let pp = new ParchmintParser(validParchmintArchBareBones);
+            let arch = pp.parse();
+
+            expect(pp.valid).toBe(true);
+            expect(arch.name).toBe('bare-bones-architecture-name');
+            expect(arch.layers.length).toBe(1);
+        });
+
+        test('method argument', () => {
+            let pp = new ParchmintParser();
+            let arch = pp.parse(validParchmintArchBareBones);
+
+            expect(pp.valid).toBe(true);
+            expect(arch.name).toBe('bare-bones-architecture-name');
+            expect(arch.layers.length).toBe(1);
+        });
+    });
+});
+
+describe('field access', () => {
+    test('architecture', () => {
+        let pp = new ParchmintParser(readme_parchmint);
+        pp.parse();
+
+        expect(pp.valid).toBe(true);
+        expect(pp.architecture).toBeTruthy();
+        expect(pp.architecture.layers.length).toBe(2);
+    });
+
+    test('valid', () => {
+        let pp = new ParchmintParser(readme_parchmint);
+        pp.parse();
+
+        expect(pp.valid).toBe(true);
+    });
+
+    test('parchmint', () => {
+        let pp = new ParchmintParser(readme_parchmint);
+
+        expect(pp.parchmint).toEqual(readme_parchmint);
+    });
+});
+
+describe('clear method', () => {
+    test('fields', () => {
+        let pp = new ParchmintParser(readme_parchmint);
+        pp.parse();
+
+        // We have to set valid false directly because I was silly and chose to use a valid parchmint
+        pp.valid = false;
```

Comment:
Can we just use a non-valid ParchMint here? This seems almost like an anti-test since we test a value we set externally.

Reply:
Changed to an non-valid Parchmint file.